File: java/src/main/java/tjchiang/leetcode/L0167_Two_Sum_II/TwoSumII.java
package tjchiang.leetcode.L0167_Two_Sum_II;

/**
 * <a href="https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/description/">leetcode</a>
 */
public class TwoSumII {
    public int[] solve(int[] numbers, int target) {
        int start = 0;
        int end = numbers.length - 1;

        while (start < end) {
            int sum = numbers[start] + numbers[end];

            if (sum == target) return new int[]{start + 1, end + 1};
            if (sum < target) start++;
            else end--;
        }
        return new int[]{};
    }
}

File: leetcode/0167-Two_Sum_II/two_sum_ii.py
from typing import List

class TwoSumII:
    def solve(self, numbers: List[int], target: int) -> List[int]:
        start: int = 0
        end: int = len(numbers) - 1
        while start < end:
            result = numbers[start] + numbers[end]
            if result == target:
                return [start + 1, end + 1]
            if result < target:
                start += 1
            else:
                end -= 1

        return [-1, -1]
